### DISCUSSION TITLE
Speed on Save

### DISCUSSION THREAD

**Discussion by arpowers** [vuejs/language-tools#144] [General]
Just trying out this extension and I love having type checking in the Vue component templates. The only problem is that on "save" Volar takes around 10 seconds to compile and do its thing. 

Am I doing something wrong? Is there a way to speed this up? 

<img width="639" alt="Screen Shot 2021-04-20 at 2 32 23 PM" src="https://user-images.githubusercontent.com/308134/115466653-3db84000-a1e5-11eb-95af-72dec11d1f10.png">

**Comment by johnsoncodehk** [comment]
Please try to disable code action on save, see: https://stackoverflow.com/questions/63992430/visual-studio-code-how-to-disable-editor-codeactionsonsave

**Comment by arpowers** [comment]
So to use this plugin basically we have to forgo all formatting/linting on save?  Is there a doc on this somewhere?

Would be helpful to get an idea of what needs to change vs Vetur since that's been the modus operandi. 

Otherwise, some very cool type checking happening in the templates with this plugin. Good job! @johnsoncodehk

**Reply by johnsoncodehk** [reply]
First thank you. :)

I guess you has config `Fix All` code action on save, Vetur do not support code actions like `Remove all unused code`, `Organize Imports`... so it just do nothing on save, but Volar support it, so it will do multiple code actions when you save file.

You can config formatting/linting on save, Volar formatting not too slow, but you should avoid `Fix All` on save.

**Reply by izhaoqing** [reply]
如果我需要使用 eslint 和 stylelint 的 fix all，可以单独关掉 volar 吗？类似于 `"source.fixAll.volar": false` 。

**Reply by jagoncalves14** [reply]
> 如果我需要使用 eslint 和 stylelint 的 fix all，可以单独关掉 volar 吗？类似于 `"source.fixAll.volar": false` 。


I have that same doubt - tomorrow I’ll try disabling Volar and leave only eslint and stylelint to make the on save actions.

I’ll leave an update if I find improvements 👌

**Reply by jlabresh1** [reply]
@jagoncalves14 did you find any improvements?

**Reply by johnsoncodehk** [reply]
Please try use `source.fixAll.eslint` instead of `source.fixAll`.

**Comment by sandercoffee** [comment]
Worked for me


````json
  "[vue]": {
    "gitlens.codeLens.scopes": ["document"],
    "editor.defaultFormatter": "esbenp.prettier-vscode",
    "editor.formatOnSave": true,
  },
  "editor.codeActionsOnSave": {
    "source.fixAll": false,
    "source.organizeImports": false
  },
````

**Reply by Miguel-Bento-Github** [reply]
Removing

```
"editor.codeActionsOnSave": ["source.organizeImports"]
```

from my global settings, did the trick. Hope it helps others.
